### PR TITLE
RavenDB-15441 - add support for LINQ for first class patch

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-15441

### CODE REVIEW

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs [hunk 1]
```diff
@@ -235,7 +235,29 @@ public async Task<ServerNode> GetCurrentSessionNode()
...
+                JavascriptConversionExtensions.ReservedWordsSupport.Instance,
+                JavascriptConversionExtensions.ValueTypeParseSupport.Instance,
+                JavascriptConversionExtensions.JsonPropertyAttributeSupport.Instance,
+                JavascriptConversionExtensions.NullComparisonSupport.Instance,
+                JavascriptConversionExtensions.NullableSupport.Instance,
+                JavascriptConversionExtensions.NewSupport.Instance,
+                JavascriptConversionExtensions.ListInitSupport.Instance,
+                MemberInitAsJson.ForAllTypes
+            };
+
```
Comment (aviv86) [inline src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs]: do we need all these extensions ?

Comment (aviv86) [reply src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs]: I see that in v5.0 we changed it to this:
https://github.com/ravendb/ravendb/blob/0c45f188594f7328a139184047a3738a53ba77a0/src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs#L232-L237

Comment (ppekrol) [reply src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs]: Let's check if this passes in 5.0. If yes. Add a test there and close down the issue. Do not want to make that change in 4.2

Comment (grisha-kotler) [reply src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs]: done

**Review by ppekrol — APPROVED** [review]

**Pull request merged**